java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -1885,4 +1959,34 @@ public CompletableFuture<String> watch(@NonNull String[] keys) {
     public CompletableFuture<String> unwatch() {
         return commandManager.submitNewCommand(UnWatch, new String[0], this::handleStringResponse);
     }
+
+    // Hack: convert all `byte[]` -> `GlideString`. Better doing it here in the Java realm
+    // rather than doing it in the Rust code using JNI calls (performance)
+    private Object convertByteArrayToGlideString(Object o) {
+        if (o == null) return o;
+
+        if (o instanceof byte[]) {
+            o = GlideString.of((byte[]) o);
+        } else if (o.getClass().isArray()) {
+            var array = (Object[]) o;
+            for (var i = 0; i < array.length; i++) {
+                array[i] = convertByteArrayToGlideString(array[i]);
+            }
+        } else if (o instanceof Set) {
+            var set = (Set<?>) o;
+            o = set.stream().map(this::convertByteArrayToGlideString).collect(Collectors.toSet());
+        } else if (o instanceof Map) {
+            var map = (Map<?, ?>) o;
+            o =
+                    map.entrySet().stream()
+                            .collect(
+                                    HashMap::new,
```

Comment:
Use `LinkedHashMap` - it guarantees the order
```suggestion
                                    LinkedHashMap::new,
```